### PR TITLE
AG-12204 - Enable ban-ts-comment eslint rule.

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-12204

### CODE REVIEW

File: packages/ag-charts-community/src/util/color.ts [hunk 1]
```diff
@@ -466,8 +466,7 @@ export class Color implements IColor {
      * CSS Color Module Level 4:
      * https://drafts.csswg.org/css-color/#named-colors
      */
-    private static readonly nameToHex: Record<string, string> = {
-        // @ts-expect-error
+    private static readonly nameToHex: Record<string, string | null> = {
         __proto__: null,
```
Comment (iMoses) [inline packages/ag-charts-community/src/util/color.ts]: nit: is `__proto__` really needed here?

Comment (alantreadway) [reply packages/ag-charts-community/src/util/color.ts]: Discussed with @jacobp100 and switched to a `Map`.

File: libraries/ag-charts-test/src/canvas/mock-canvas.ts [hunk 2]
```diff
@@ -31,14 +31,13 @@ Object.defineProperty(NodeCanvasRenderingContext2D.prototype, 'createConicGradie
 // https://github.com/Automattic/node-canvas/issues/1852
 const context2dTransform = NodeCanvasRenderingContext2D.prototype.transform;
 Object.defineProperty(NodeCanvasRenderingContext2D.prototype, 'transform', {
-    value: function transform(...args: any[]) {
-        if (args[0] === 0) {
+    value: function transform(a: number, b: number, c: number, d: number, e: number, f: number) {
```
Comment (jacobp100) [inline libraries/ag-charts-test/src/canvas/mock-canvas.ts:34]: You can also pass in a single DOMMatrix - so I'm not sure this is safe

File: packages/ag-charts-enterprise/src/axes/polar-crosslines/test/examples.ts [hunk 3]
```diff
@@ -273,7 +273,7 @@ export const RADIAL_BAR_POLYGON_CROSSLINES: Options = {
         axes: [
             {
                 type: 'radius-category',
-                // @ts-expect-error
+                // @ts-expect-error undocumented option
```
Comment (jacobp100) [inline packages/ag-charts-enterprise/src/axes/polar-crosslines/test/examples.ts]: Should this not be documented?

**Review by jacobp100 — APPROVED** [review]

File: packages/ag-charts-enterprise/src/axes/polar-crosslines/test/examples.ts (hunk 3 above)
Comment (alantreadway) [reply packages/ag-charts-enterprise/src/axes/polar-crosslines/test/examples.ts]: The feature itself doesn't really make any sense - I've removed this example + test-case.

File: libraries/ag-charts-test/src/canvas/mock-canvas.ts (hunk 2 above)
Comment (alantreadway) [reply libraries/ag-charts-test/src/canvas/mock-canvas.ts:34]: Discussed offline.

**Pull request merged**